lib/semantics/scope.cc

```diff
@@ -185,12 +185,17 @@ bool Scope::CanImport(const SourceName &name) const {
   }
 }
 
-const Scope *Scope::FindScope(const parser::CharBlock &source) const {
+const Scope *Scope::FindScope(parser::CharBlock source) const {
+  return const_cast<const Scope *>(
+      const_cast<Scope *>(this)->FindScope(source));
+}
+
+Scope *Scope::FindScope(parser::CharBlock source) {
```

Comment:
Why change to passing source by value?

Reply:
CharBlocks are tiny values and the value semantics are more clear than const references, so I often make this change whenever I notice the usage.  Like an int, a CharBlock is just a context-free value.  Using a const reference to a simple value raises questions like "why does this have to be in memory?" and "will its address be taken?  do I have to worry about lifetimes?".